tests/PostgreSQLSchemaManagerTest.php

```diff
@@ -0,0 +1,138 @@
+<?php
+
+namespace SilverStripe\PostgreSQL\Tests;
+
+use SilverStripe\Core\ClassInfo;
+use SilverStripe\Dev\SapphireTest;
+use SilverStripe\ORM\Connect\Database;
+use SilverStripe\ORM\Connect\DatabaseException;
+use SilverStripe\ORM\DB;
+use SilverStripe\PostgreSQL\PostgreSQLConnector;
+use SilverStripe\PostgreSQL\PostgreSQLSchemaManager;
+
+class PostgreSQLSchemaManagerTest extends SapphireTest
+{
+
+    protected $usesTransactions = false;
+
+    public function testAlterTable()
+    {
+        try {
+            /** @var PostgreSQLSchemaManager $dbSchema */
+            $dbSchema = DB::get_schema();
+            $dbSchema->quiet();
+
+            $this->createSS3Table();
+
+            try {
+                DB::query('INSERT INTO "ClassNamesUpgrade" ("ClassName") VALUES (\'App\MySite\FooBar\')');
+                $this->assertFalse(true, 'SS3 Constaint should have blocked the previous insert.');
+            } catch (DatabaseException $ex) { }
+
+            $dbSchema->schemaUpdate(function () use ($dbSchema) {
+                $dbSchema->requireTable(
+                    'ClassNamesUpgrade',
+                    [
+                        'ID' => 'PrimaryKey',
+                        'ClassName' => 'Enum(array("App\\\\MySite\\\\FooBar"))',
+                    ]
+                );
+            });
+
+            DB::query('INSERT INTO "ClassNamesUpgrade" ("ClassName") VALUES (\'App\MySite\FooBar\')');
```

Comment:
At this point I'm getting the following constraint:
```
\d "ClassNamesUpgrade"
                                   Table "public.ClassNamesUpgrade"
  Column   |          Type          |                            Modifiers                             
-----------+------------------------+------------------------------------------------------------------
 ID        | bigint                 | not null default nextval('"ClassNamesUpgrade_ID_seq"'::regclass)
 ClassName | character varying(255) | default 'App\MySite\FooBar'::character varying
Indexes:
    "ClassNamesUpgrade_pkey" PRIMARY KEY, btree ("ID")
Check constraints:
    "ClassNamesUpgrade_ClassName_check" CHECK ("ClassName"::text = ANY (ARRAY['App\MySite\FooBar'::character varying, NULL::character varying]::text[]))
```

We're getting an extra NULL entry there, which basically disables the constraint.
Feels like a regression, although might have been introduced not in this PR - I'll have a look around.

Reply:
I've tracked it down to this bit of code: https://github.com/silverstripe/silverstripe-postgresql/blob/685e33cf844a48d08f9587c5758867fb17f5b6cb/code/PostgreSQLSchemaManager.php#L1105

I reckon we should change it to `check ("fld" in ('a', 'b') or "fld" is null)`

Reply:
However, that will require patching the regex above as well.

Reply:
It's arguably a regression on the SS3 version of the module however, that's what the current SS4 module does. So this problem wasn't introduced by this PR.

Another consideration is that we'll need to provide a migration path not only for people upgrading for the SS3 version of the module, but also people upgrading from an SS4 version that doesn't enforce enum column data integrity.

I think it would be better to consider this as a separate issue https://github.com/silverstripe/silverstripe-postgresql/issues/105

Reply:
Agreed